app/Main.hs

```diff
@@ -0,0 +1,369 @@
+    ( MkPayment (..) )
+import Data.UTxO.Transaction.Cardano.Byron
+    ( Byron
+    , decodeCoinSel
+    , decodeTx
+    , encodeCoinSel
+    , encodeTx
+    , fromBase16
+    , fromBase58
+    , mainnetMagic
+    , mkInit
+    , mkInput
+    , mkOutput
+    , mkSignKey
+    , testnetMagic
+    )
+import Data.Word
+    ( Word32 )
+import Numeric.Natural
+    ( Natural )
+import Options.Applicative.Help.Pretty
+    ( hardline, indent, string, vsep )
+import System.Console.ANSI
+    ( Color (..)
+    , ColorIntensity (..)
+    , ConsoleLayer (..)
+    , SGR (..)
+    , hSetSGR
+    , hSupportsANSIWithoutEmulation
+    )
+import System.Exit
+    ( exitFailure )
+import System.IO
+    ( Handle, hIsTerminalDevice, stderr, stdin, stdout )
+
+import Options.Applicative
+
+import qualified Codec.CBOR.Decoding as CBOR
+import qualified Codec.CBOR.Encoding as CBOR
+import qualified Codec.CBOR.Read as CBOR
+import qualified Codec.CBOR.Write as CBOR
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as BL
+import qualified Data.Text as T
+import qualified Data.Text.Encoding as T
+import qualified Data.Text.IO as TIO
+import qualified Data.UTxO.Transaction as Tx
+
+
+main :: IO ()
+main = setup >> parseCmd >>= \case
+    CmdEmpty Empty{protocolMagic} -> do
+        let state0 = Tx.empty protocolMagic
+        hPutState stdout (encodeCoinSel state0)
+
+    CmdAddInput AddInput{inputIx,inputTxId} -> do
+        state <- hGetState stdin decodeCoinSel
+        case mkInput inputIx inputTxId of
+            Nothing -> failWith "Invalid input index or transaction id."
+            Just input -> do
+                let state' = Tx.addInput input state
+                hPutState stdout (encodeCoinSel state')
+
+    CmdAddOutput AddOutput{coin,address} -> do
+        state <- hGetState stdin decodeCoinSel
+        case mkOutput coin address of
+            Nothing -> failWith "Invalid output value or address."
+            Just output -> do
+                let state' = Tx.addOutput output state
+                hPutState stdout (encodeCoinSel state')
+
+    CmdLock -> do
+        state <- hGetState stdin decodeCoinSel
+        let state' = Tx.lock state
+        hPutState stdout (encodeTx state')
+
+    CmdSignWith SignWith{prvKey} -> do
+        state <- hGetState stdin decodeTx
+        case mkSignKey prvKey of
+            Nothing -> failWith "Invalid signing key."
+            Just signKey -> do
+                let state' = Tx.signWith signKey state
+                hPutState stdout (encodeTx state')
+
+    CmdSerialize -> do
+        state <- hGetState stdin decodeTx
+        case Tx.serialize state of
+            Left e -> failWith (show e)
+            Right bytes -> TIO.putStr (base64 bytes)
+  where
+    setup :: IO ()
+    setup = do
+        -- Enable ANSI colors on Windows.
+        void $ hSupportsANSIWithoutEmulation stderr
+
+parseCmd :: IO Cmd
+parseCmd = customExecParser (prefs showHelpOnEmpty) cmd
```

Comment:
we are satisfied with output when for example 
```
cardano-tx blabla
```
?

Reply:
```
$ cardano-tx blabla
Invalid argument `blabla'

Usage: cardano-tx COMMAND
  cardano-tx
```

:thinking: .. could be slightly better but, not bad :no_mouth: ... I'll see if we can tweak the preferences to also show the help.